src/Control/Lens/Internal/PrismTH.hs

```diff
@@ -105,6 +105,31 @@ makePrisms = makePrisms' True
 -- Generate an "As" class of prisms. Names are selected by prefixing the constructor
 -- name with an underscore.  Constructors with multiple fields will
 -- construct Prisms to tuples of those fields.
+--
+-- In the event that the name of a data type is also the name of one of its
+-- constructors, the name of the 'Prism' generated for the data type will be
+-- prefixed with an extra @_@ (if the data type name is prefix) or @.@ (if the
+-- name is infix) to disambiguate it from the 'Prism' for the corresponding
+-- constructor. For example, this code:
+--
+-- @
+-- data Quux = Quux Int | Quuz Bool
```

Comment:
These are dyslexia unfriendly, I read that as `Quux Int | Quux Bool`; maybe `Quux Int | Fuzzy Bool` ?

Reply:
I somewhat deliberately chose the metasyntactic variable name `Quuz` so that: it would have the same length as `Quux`, ensuring that the spacing of their type signatures would align nicely. I'd certainly be willing to switch it out with a different variable name of the same length. Some potential candidates are `Fred` and `Thud` (taken from [here](https://tools.ietf.org/html/rfc3092)).

Reply:
I went ahead and swapped out `Quuz` for `Fred`.

Reply:
Thanks!